deployer/pom.xml

```diff
@@ -16,23 +16,118 @@
+            <artifactId>brooklyn-test-support</artifactId>
+            <version>${brooklyn.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.brooklyn</groupId>
+            <artifactId>brooklyn-core</artifactId>
+            <version>${brooklyn.version}</version>
+            <classifier>tests</classifier>
+            <scope>test</scope>
+        </dependency>
 
-  <groupId>eu.seaclouds-project</groupId>
-  <artifactId>deployer</artifactId>
-  <packaging>jar</packaging>
+        <dependency>
+            <groupId>org.apache.brooklyn</groupId>
+            <artifactId>brooklyn-launcher</artifactId>
+            <version>${brooklyn.version}</version>
+            <scope>test</scope>
+        </dependency>
 
-  <name>deployer</name>
-  <url>http://maven.apache.org</url>
+        <dependency>
+            <!-- this gives us flexible and easy-to-use logging; just edit logback-custom.xml! -->
+            <groupId>org.apache.brooklyn</groupId>
+            <artifactId>brooklyn-logback-xml</artifactId>
+            <version>${brooklyn.version}</version>
+            <!-- optional so that this project has logging; dependencies may redeclare or supply their own;
+                 provided so that it isn't put into the assembly (as it supplies its own explicit logback.xml);
+                 see Logging in the Brooklyn website/userguide for more info -->
+            <optional>true</optional>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
 
-  <properties>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-  </properties>
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <configuration>
+                    <descriptors>
+                        <descriptor>src/main/assembly/assembly.xml</descriptor>
+                    </descriptors>
+                    <appendAssemblyId>false</appendAssemblyId>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>make-assembly</id>
+                        <!-- this is used for inheritance merges -->
+                        <phase>package</phase>
+                        <!-- bind to the packaging phase -->
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
 
+            <plugin>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>2.3.1</version>
+                <executions>
+                    <execution>
+                        <id>default-jar</id>
+                        <!-- put the default-jar in the none phase to skip it from being created -->
+                        <phase>none</phase>
```

Comment:
I see your point, although not a big fan, I think you should consider
maven-shade-plugin to create a uber jar, with all the necessary includes.
This the module will produce a simple jar and a uber-jar to be used in the
yaml blueprint. Like it?

Il giorno mar 21 lug 2015 19:07 Jose <notifications@github.com> ha scritto:

> In deployer/pom.xml
> <https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/82#discussion_r35127329>
> :
>
> >
> > +            <plugin>
> > +                <artifactId>maven-jar-plugin</artifactId>
> > +                <version>2.3.1</version>
> > +                <executions>
> > +                    <execution>
> > +                        <id>default-jar</id>
> > +                        <!-- put the default-jar in the none phase to skip it from being created -->
> > +                        <phase>none</phase>
>
> @andreaturli <https://github.com/andreaturli> I have deleted the deployer
> assembly and I have added all dependencies as resources to
> installer/pom.xml.
> As I have mentioned above, I think that maintain this approach will be an
> error prone task, because we will have to maintain the integrity between
> the different dependencies versions if we modified any version.
> If we add a new dependency, such as OpenShift client, we will add this
> dependency and its transitive dependencies as resource and toBrooklyNode
> classpath.
>
> Moreover, I think that the number of dependencies that we need to add to
> the BrooklynNode classpath is too big.
> Then, from my point of view I think it is easier the deployer assembly.
> However, it is just my opinion, the approach of this version works
> correctly so we could be to use it. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/82/files#r35127329>
> .
>

-- 
Cloudsoft Corporation Limited, Registered in Scotland No: SC349230. 
 Registered Office: 13 Dryden Place, Edinburgh, EH9 1RP
 
This e-mail message is confidential and for use by the addressee only. If 
the message is received by anyone other than the addressee, please return 
the message to the sender by replying to it and then delete the message 
from your computer. Internet e-mails are not necessarily secure. Cloudsoft 
Corporation Limited does not accept responsibility for changes made to this 
message after it was sent.

Whilst all reasonable care has been taken to avoid the transmission of 
viruses, it is the responsibility of the recipient to ensure that the 
onward transmission, opening or use of this message and any attachments 
will not adversely affect its systems or data. No responsibility is 
accepted by Cloudsoft Corporation Limited in this regard and the recipient 
should carry out such virus and other checks as it considers appropriate.